tests/prompts/parser/test_parser.py

```diff
@@ -316,6 +316,17 @@ def test_variant_delimiter(self, parser: Parser):
 
         sequence = parser.parse("{2$$  $$cat|dog|bird}")
 
+    def test_range_sd_issue_223(self, parser: Parser, prompt: str):
+        # > {0-1$$a|b|c|d} would return nothing or one item. 0-1 could also be 0-3 etc.
+        # > Since 2.52, if the random number picker lands on 0, instead of returning an empty set,
+        # > the parser just stops and kicks out whatever it has, which results in a broken prompt.
+        # https://github.com/adieyal/sd-dynamic-prompts/issues/223
+        sequence = parser.parse(r"{0-1$$a|b|c|d}")
+        var = sequence[0]
+        assert isinstance(var, VariantCommand)
+        assert var.min_bound == 0
+        assert var.max_bound == 1
+
```

Comment:
have you tested it with the {Random,Combinatorial}PromptGenerators in case they don't handle zero results?

Reply:
I'll check when I can, later. Maybe we should have a smoke test suite with a bunch of usual prompts...

Reply:
Not a bad idea - it would have to be at the generator level I  imagine in order to do a complete end-to-end. I don't think we can get away from mocking random from RandomPromptGenerator unless you think there is another way to test it?

Reply:
For a smoke test, it would be enough to see that nothing outright crashes with any given input. I went ahead and added a second commit that does test this prompt with both generators.